pkg/kubelet/kubelet.go

```diff
@@ -2972,22 +3070,47 @@ func (kl *Kubelet) CheckpointContainer(
 		return fmt.Errorf("container %v not found", containerName)
 	}
 
-	options.Location = filepath.Join(
+	checkpointDirectory := filepath.Join(
 		kl.getCheckpointsDir(),
+		string(podUID),
+	)
+
+	checkPointDirectoryInfo, err := os.Lstat(checkpointDirectory)
+	if err == nil && !checkPointDirectoryInfo.Mode().IsDir() {
+		return fmt.Errorf("checkpoint destination %s needs to be a directory", checkpointDirectory)
+	} else if err != nil && !os.IsNotExist(err) {
+		return fmt.Errorf("error stating checkpoint destination %s:%w", checkpointDirectory, err)
+	} else if err != nil && os.IsNotExist(err) {
+		if err = os.Mkdir(checkpointDirectory, 0700); err != nil {
+			return fmt.Errorf("error creating checkpoint directory %s:%w", checkpointDirectory, err)
+		}
+	}
+
+	kl.maxCheckpointsPerContainerLock.Lock()
+	options.Location = filepath.Join(
+		checkpointDirectory,
 		fmt.Sprintf(
-			"checkpoint-%s-%s-%s.tar",
+			"checkpoint-%s-%s-%d-%0*d.tar",
 			podFullName,
 			containerName,
-			time.Now().Format(time.RFC3339),
+			time.Now().UnixNano(),
+			int(math.Log10(float64(math.MaxInt32)))+1,
```

Comment:
? pls add a comment explaining this ..

Reply:
Yes, for sorting the counter should always include all leading zeroes. I can change this to 10. I just didn't want to count it manually and let the computer do the work.

Reply:
thx